File: src/main/java/com/my/admin/app/commerce/service/ExceptionService.java
package com.my.admin.app.commerce.service;

import com.black.food.manager.exception.ApplicationException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Component;

@Component
public class ExceptionService {
    private static final Logger LOGGER = LoggerFactory.getLogger(ExceptionService.class);

    public RuntimeException throwNotFoundException(final String message) {
        LOGGER.error(message);
        throw new ApplicationException(message, HttpStatus.NOT_FOUND);
    }

    public RuntimeException throwConflictException(final String message) {
        LOGGER.error(message);
        throw new ApplicationException(message, HttpStatus.CONFLICT);
    }
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/service/ExceptionService.kt
package com.bk.reactive.app.commerce.admin.my.service

import com.bk.reactive.app.commerce.admin.my.exception.api.ApplicationException
import mu.KotlinLogging
import org.springframework.http.HttpStatus
import org.springframework.stereotype.Service

@Service
class ExceptionService {
    private val logger = KotlinLogging.logger {}

    fun throwNotFoundException(message: String): RuntimeException {
        logger.error { message }
        throw ApplicationException(
            message = message,
            code = HttpStatus.NOT_FOUND,
            cause = Throwable(message)
        )
    }

    fun throwNotFoundException(id: String, param: String): ApplicationException {
        val message = "Document with $param: $id not found"
        logger.error { message }
        return ApplicationException(
            message = message,
            code = HttpStatus.NOT_FOUND,
            cause = Throwable(message)
        )
    }

    fun throwConflictException(message: String): RuntimeException {
        logger.error { message }
        throw ApplicationException(
            message = message,
            code = HttpStatus.CONFLICT,
            cause = Throwable(message)
        )
    }

    fun throwNotFoundExceptionToCashStatus(id: String, param: String): RuntimeException {
        val message = "Cash status with $param $id not found"
        logger.error { message }
        throw ApplicationException(
            message = message,
            code = HttpStatus.NOT_FOUND,
            cause = Throwable(message)
        )
    }

    fun throwConflictException(message: String, error: String): RuntimeException {
        val completeMessage = "Error : $message"
        logger.error { error }
        logger.error { completeMessage }
        throw ApplicationException(
            message = completeMessage,
            code = HttpStatus.CONFLICT,
            cause = Throwable(message)
        )
    }
}